tests/functionaltests/Tests/ToastNotificationTests.mm

```diff
@@ -0,0 +1,370 @@
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <COMIncludes.h>
+#include "MockClass.h"
+#include <windows.applicationModel.activation.h>
+#include <COMIncludes_End.h>
+
+#include <UWP/WindowsApplicationModelActivation.h>
+#include <TestFramework.h>
+#include <Foundation/Foundation.h>
+#include "UIKit/UIApplication.h"
+#include "StringHelpers.h"
+
+using namespace ABI::Windows::ApplicationModel::Activation;
+using namespace ABI::Windows::Foundation::Collections;
+using namespace ABI::Windows::Foundation;
+using namespace Microsoft::WRL;
+
+// Method to call in tests to activate app
+extern "C" void UIApplicationActivationTest(IInspectable* args, void* delegateClassName);
+
+MOCK_CLASS(MockPropertyValue,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IPropertyValue> {
+
+           public:
+               MOCK_STDCALL_METHOD_1(get_Type);
+               MOCK_STDCALL_METHOD_1(get_IsNumericScalar);
+               MOCK_STDCALL_METHOD_1(GetUInt8);
+               MOCK_STDCALL_METHOD_1(GetInt16);
+               MOCK_STDCALL_METHOD_1(GetUInt16);
+               MOCK_STDCALL_METHOD_1(GetInt32);
+               MOCK_STDCALL_METHOD_1(GetUInt32);
+               MOCK_STDCALL_METHOD_1(GetInt64);
+               MOCK_STDCALL_METHOD_1(GetUInt64);
+               MOCK_STDCALL_METHOD_1(GetSingle);
+               MOCK_STDCALL_METHOD_1(GetDouble);
+               MOCK_STDCALL_METHOD_1(GetChar16);
+               MOCK_STDCALL_METHOD_1(GetBoolean);
+               MOCK_STDCALL_METHOD_1(GetString);
+               MOCK_STDCALL_METHOD_1(GetGuid);
+               MOCK_STDCALL_METHOD_1(GetDateTime);
+               MOCK_STDCALL_METHOD_1(GetTimeSpan);
+               MOCK_STDCALL_METHOD_1(GetPoint);
+               MOCK_STDCALL_METHOD_1(GetSize);
+               MOCK_STDCALL_METHOD_1(GetRect);
+               MOCK_STDCALL_METHOD_2(GetUInt8Array);
+               MOCK_STDCALL_METHOD_2(GetInt16Array);
+               MOCK_STDCALL_METHOD_2(GetUInt16Array);
+               MOCK_STDCALL_METHOD_2(GetInt32Array);
+               MOCK_STDCALL_METHOD_2(GetUInt32Array);
+               MOCK_STDCALL_METHOD_2(GetInt64Array);
+               MOCK_STDCALL_METHOD_2(GetUInt64Array);
+               MOCK_STDCALL_METHOD_2(GetSingleArray);
+               MOCK_STDCALL_METHOD_2(GetDoubleArray);
+               MOCK_STDCALL_METHOD_2(GetChar16Array);
+               MOCK_STDCALL_METHOD_2(GetBooleanArray);
+               MOCK_STDCALL_METHOD_2(GetStringArray);
+               MOCK_STDCALL_METHOD_2(GetInspectableArray);
+               MOCK_STDCALL_METHOD_2(GetGuidArray);
+               MOCK_STDCALL_METHOD_2(GetDateTimeArray);
+               MOCK_STDCALL_METHOD_2(GetTimeSpanArray);
+               MOCK_STDCALL_METHOD_2(GetPointArray);
+               MOCK_STDCALL_METHOD_2(GetSizeArray);
+               MOCK_STDCALL_METHOD_2(GetRectArray);
+           });
```

Comment:
seeing as a property value is just a dumb container anyway, I don't see the value in mocking it ...